contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,260 @@
+    IPool public immutable pool;
+
+    /// @notice reference to the Maple Staking Rewards Contract
+    IMplRewards public immutable mplRewards;
+
+    /// @notice reference to the underlying token
+    IERC20 public immutable token;
+
+    /// @notice reference to the Maple token
+    IERC20 public immutable rewardsToken;
+
+    /// @notice scaling factor for USDC
+    /// @dev hardcoded to use USDC decimals as this is the only
+    /// supplied asset Volt Protocol will support
+    uint256 public constant scalingFactor = 1e12;
+
+    /// @notice fetch underlying asset by calling pool and getting liquidity asset
+    /// @param _core reference to the Core contract
+    /// @param _pool Maple Pool contract
+    /// @param _mplRewards Maple Rewards contract
+    constructor(
+        address _core,
+        address _pool,
+        address _mplRewards
+    ) CoreRef(_core) {
+        token = IERC20(IPool(_pool).liquidityAsset());
+        /// enforce underlying token is USDC
+        require(
+            address(token) == 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,
+            "MaplePCVDeposit: Underlying not USDC"
+        );
+        pool = IPool(_pool);
+        mplRewards = IMplRewards(_mplRewards);
+        rewardsToken = IERC20(IMplRewards(_mplRewards).rewardsToken());
+    }
+
+    /// @notice return the amount of funds this contract owns in USDC
+    /// accounting for interest earned
+    /// and unrealized losses in the venue
+    function balance() public view override returns (uint256) {
+        uint256 rawBalance = pool.balanceOf(address(this)) +
+            pool.accumulativeFundsOf(address(this)) -
+            pool.recognizableLossesOf(address(this));
+        return rawBalance / scalingFactor;
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return address(token);
+    }
+
+    /// ---------- Happy Path APIs ----------
+
+    /// @notice deposit PCV into Maple.
+    /// all deposits are subject to a minimum 90 day lockup,
+    /// no op if 0 token balance
+    /// deposits are then immediately staked to accrue MPL rewards
+    /// only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        /// pool deposit
+        token.approve(address(pool), amount);
+        pool.deposit(amount);
+
+        /// stake pool FDT for MPL rewards
+        uint256 scaledDepositAmount = amount * scalingFactor;
+        pool.increaseCustodyAllowance(address(mplRewards), scaledDepositAmount);
+        mplRewards.stake(scaledDepositAmount);
```

Comment:
the `MplRewards` contract is pausable, should we check here whether it's paused or not before staking